org.adempiere.base/src/org/compiere/model/MTableIndex.java

```diff
@@ -117,10 +119,31 @@ public MTableIndex(MTable parent, String name)
 	 */
 	public MIndexColumn[] getColumns(boolean reload)
 	{
-		if (m_columns != null && !reload)
+		String whereClause = MIndexColumn.COLUMNNAME_AD_TableIndex_ID + "=?";
+		return getColumns(reload, whereClause);

```

Comment:
this should be simplify to just return getColumns(reload, false);

Reply:
done.